corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js

```diff
@@ -0,0 +1,97 @@
+$(function(){
+    var PAD_CHAR = '&nbsp;';
+    function ToggleView() {
+        var self = this;
+        self.items = ko.observableArray();
+
+        self.init = function (config) {
+            self.padded_ns = {};
+            var max_ns_len = Math.max.apply(Math, _.map(config.namespaces, function (ns) { return ns.length }));
+            _(config.namespaces).each(function (namespace) {
+                var diff = max_ns_len - namespace.length,
+                    pad = new Array(diff + 1).join(PAD_CHAR);
+                self.padded_ns[namespace] = namespace + pad;
+            });
+            self.init_items(config);
+            self.latest_use = ko.observable(config.last_used._latest || '');
+        };
+
+        self.init_items = function (config) {
+            var items = config.items,
+                last_used = config.last_used || {};
+            self.items.removeAll();
+            _(items).each(function (item) {
+                var fields = item.split(':'),
+                    namespace = fields.length > 1 ? fields[0] : 'user',
+                    value = fields.length > 1 ? fields[1] : fields[0];
+                self.items.push({
+                    namespace: ko.observable(self.padded_ns[namespace]),
+                    value: ko.observable(value),
+                    last_used: ko.observable(last_used[value])
+                });
+            });
+        };
+
+        self.addItem = function (namespace) {
+            self.items.push({
+                namespace: ko.observable(self.padded_ns[namespace]),
+                value: ko.observable(),
+                last_used: ko.observable()
+            });
```

Comment:
Not completely sure how `observable` behaves. Was thinking something like
```
self.pushItem = function(namespace, value, last_used)
```
and then calling self.pushItem(namespace, null_or_some_default_value, {}_or_some_default_value) for the above case.
Can ignore this if thats not feasible.